iml-agent/src/action_plugins/lamigo.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{agent_error::ImlAgentError, env};
+use futures::future::TryFutureExt;
+use std::collections::HashMap;
+use std::fmt;
+use std::path::Path;
+use tokio::fs;
+use tokio::io::AsyncWriteExt;
+
+#[derive(serde::Deserialize, structopt::StructOpt, Clone, Debug)]
+pub struct Config {
+    #[structopt(long)]
+    /// File system name
+    fs: String,
+
+    #[structopt(long)]
+    /// MDT device that provides changelogs
+    mdt: String,
+
+    #[structopt(long)]
+    /// The user on which behalf lamigo consumes changelogs from the MDT device
+    user: String,
+
+    #[structopt(long)]
+    /// Fast OST pool name
+    hot_pool: String,
+
+    #[structopt(long)]
+    /// Slow OST pool name
+    cold_pool: String,
+
+    #[structopt(long)]
+    /// Interval for lamigo to wait before replicating a closed file (in seconds)
+    min_age: u32,
+
+    #[structopt(long)]
+    /// Lustre client mount point, e.g. `/mnt/lustre`
+    mount_point: String,
+
+    #[structopt(long)]
+    /// Lustre device to be mounted, e.g. `192.168.0.100@tcp0:/spfs`
+    lustre_device: String,
+}
+
+impl fmt::Display for Config {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "[Unit]\n\
+             Description=Run lamigo service\n\
+             [Service]\n\
+             ExecStart=/usr/bin/lamigo \
+             -m {fs}-{mdt} \
+             -u {user} \
+             -s {hot_pool} \
+             -t {cold_pool} \
+             -a {min_age} \
+             {mount_point}\n\
```

Comment:
We want lamigo to dump FIDs to socket (via postman) (c.f. EX-731)  The option is `--iml-socket`.  Do we need the mountpoint at all?

Reply:
Done, I'm using the option `--iml-socket` instead